backend/src/test/java/com/redhat/cloud/notifications/routers/EventServiceTest.java

```diff
@@ -75,7 +78,27 @@
     MockServerClientConfig mockServerConfig;
 
     // A new instance is automatically created by JUnit before each test is executed.
-    private ModelInstancesHolder model = new ModelInstancesHolder();
+    private final ModelInstancesHolder model = new ModelInstancesHolder();
+
+    @Test
+    void shouldNotBeAllowedToGetEventLogsWhenUserHasWrongAccessRights() {
+        Header noAccessIdentityHeader = mockRbac("tenant", "user", WRONG_ACCESS);
```

Comment:
I suspect the `mockRbac` call is not resetting the previous access. That would explain why you get `200` in the tests. Could you update it so that we have a `beforeEach` that resets the `MockServerClientConfig` instance on every call?

@gwenneg WDYT?

Reply:
I think you're right. All tests share the same `mockserver` container.

Calling [MockServerClientConfig#clearRbac](https://github.com/RedHatInsights/notifications-backend/blob/1af499740a5478949819353b025851ab347102bf/backend/src/test/java/com/redhat/cloud/notifications/MockServerClientConfig.java#L55) before each test will probably fix the issue. Or you could pass a different username to `mockRbac` when the permissions change.

Reply:
Adding a beforeEach and clearRbac() call didn't work, using a different username works. I think we can move forward with it, but it's still a workaround and we could lead to issues like these again.